dashboard/src/actions/tocActions.js

```diff
@@ -56,6 +56,14 @@ const setOptions = (data, isParent, keyPath, isDirectory) => {
     options.forEach((opt) => {
       opt["children"] = [];
     });
+  } else {
+    data.forEach((item) => {
+      options.forEach((opt) => {
+        if (item.name === opt.name) {
+          opt["size"] = item.size;
+        }
+      });
+    });
```

Comment:
This nested loop construct is really kind of disappointing.  At the very least, I would recommend limiting the traversal of the inner loop, perhaps something like this:
```javascript
    options.forEach((opt) => {
      const item = data.find((item) => item.name === opt.name);
      opt["size"] = item.size;
    });
```
In this, the primary traversal is of the `options` array, which better parallels the code at line 56, and it only traverses the `data` array as far as is required to find the item (rather than always traversing the whole thing).

Nevertheless, the fact that we're traversing the arrays twice is kind of disappointing, and the fact that, for files, we're doing an "n-squared" traversal is extra disappointing...to the point that I'm wondering if it might be better doing it all in the initial traversal, something like this:
```javascript
  const options = data.map((item) => ({
    name: item.name,
    id: isParent ? `${keyPath}*${item.name}` : item.name,
    isDirectory,
    uri: item.uri,
    [isDirectory ? "children" : "size"]: isDirectory ? [] : item.size,
  }));
```
This would allow us to omit the subsequent traversals.  (Gotta love a good ternary...or even two!  😆)

Reply:
Ah, right.  (I got caught up in the "expression-ness".)  So, something like this would be better:
```javascript
  const options = data.map((item) => {
    const option = {
      name: item.name,
      id: isParent ? `${keyPath}*${item.name}` : item.name,
      isDirectory,
      uri: item.uri,
    };
    if (isDirectory) {
      option.children = [];
    } else {
      option.size = item.size;
    }
    return option;
  });
```

Reply:
Updated the code